pkg/controller/deployment/deployment_controller.go

```diff
@@ -557,9 +597,23 @@ func (dc *DeploymentController) getNewRC(deployment extensions.Deployment) (*api
 		dc.rcExpectations.DeleteExpectations(dKey)
 		return nil, fmt.Errorf("error creating replication controller: %v", err)
 	}
+
+	if err = dc.updateDeploymentVersion(deployment, newVersion); err != nil {
+		return createdRC, err
+	}
+
 	return createdRC, nil
 }
 
+func (dc *DeploymentController) updateDeploymentVersion(deployment extensions.Deployment, version string) error {
+	if deployment.Annotations == nil {
+		deployment.Annotations = make(map[string]string)
+	}
+	deployment.Annotations[deploymentutil.VersionAnnotation] = version
```

Comment:
The reason we added generation/observedGeneration was to determine when the controller manager had observed replicas changes.

https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/controller/strategy.go#L73
	// Any changes to the spec increment the generation number, any changes to the
	// status should reflect the generation number of the corresponding object. We push
	// the burden of managing the status onto the clients because we can't (in general)
	// know here what version of spec the writer of the status has seen. It may seem like
	// we can at first -- since obj contains spec -- but in the future we will probably make
	// status its own object, and even if we don't, writes may be the result of a
	// read-update-write loop, so the contents of spec may not actually be the spec that
	// the controller has *seen*.
	//
	// TODO: Any changes to a part of the object that represents desired state (labels,
	// annotations etc) should also increment the generation.
	if !reflect.DeepEqual(oldController.Spec, newController.Spec) {
		newController.Generation = oldController.Generation + 1
	}